### ISSUE TITLE
Integrate Fig with Linux

### ISSUE THREAD

**Issue by brendanfalk** [withfig/fig#34]
Fig currently works on macOS only. We will build a port that integrates with Linux soon.

Please subscribe to this issue to get an update when Fig's autocomplete integration with the Linux is ready :smile:

**Comment by YU5Z8X2CvH1fv4ep** [comment]
4 months passed

**Comment by Alonely0** [comment]
@YU5Z8X2CvH1fv4ep And probably more months will pass. They're rewriting the entire Fig core from Swift to Rust, give them time.

**Comment by deathtrip** [comment]
Swift also supports linux, so what's wrong with it? Or are the devs just following the trend of rewriting everything in rust?

**Comment by YU5Z8X2CvH1fv4ep** [comment]
> Swift also supports linux, so what's wrong with it? Or are the devs just following the trend of rewriting everything in rust?



> @YU5Z8X2CvH1fv4ep And probably more months will pass. They're rewriting the entire Fig core from Swift to Rust, give them time.

Auto-completion in ssh without fig is wonderful. If you can tell me the principle, I may try to develop one by myself. It is too painful to wait for you to develop it.

**Comment by vieko** [comment]
Congrats on the work done so far! I had a chance to play with Fig on OSX, and it was wonderful. Looking forward to your Linux release whenever you are ready ❤️

**Comment by MostHated** [comment]
I know the original post said "Soon"(TM), but is there any chance that we might see some news on this in the near future?

**Comment by Alonely0** [comment]
I've been proposing some implementation ideas for other OSes which actually solve a lot of fig's bugs and implementarion problems in the discord. Check the thread i created on #cross-platform there

**Comment by wenealves10** [comment]
# @vieko 

Do you have any predictions for the release of fig for linux?

**Comment by wenealves10** [comment]
> Congrats on the work done so far! I had a chance to play with Fig on OSX, and it was wonderful. Looking forward to your Linux release whenever you are ready heart

Do you have any predictions for the release of fig for linux?

**Comment by zedtux** [comment]
2022 is here now, what is the status of this issue please?

**Comment by brendanfalk** [comment]
We are prioritising Linux and the Fig.js API after launching our next product in mid Feb. So aiming for an early April launch date. Stay tuned!

**Comment by tesh254** [comment]
> We are prioritising Linux and the Fig.js API after launching our next product in mid Feb. So aiming for an early April launch date. Stay tuned!

Calendar marked

**Comment by ilovefreesw** [comment]
Noice to see so many people rooting for Fig for Linux <3

**Comment by bjvalmaseda92** [comment]
> > We are prioritising Linux and the Fig.js API after launching our next product in mid Feb. So aiming for an early April launch date. Stay tuned!
> 
> 
> 
> Calendar marked

Me too 🎉😅

**Comment by opensas** [comment]
> We are prioritising Linux and the Fig.js API after launching our next product in mid Feb. So aiming for an early April launch date. Stay tuned!

That's some really awesome news! I hope you can show some proptotype or at least blog about it pretty soon

**Comment by gustavoabell** [comment]
Add support for hyper terminal with fig

**Comment by mschrage** [comment]
Some exciting progress to share! 

https://user-images.githubusercontent.com/3155109/158873604-04105ab9-4cf9-4703-8b2c-a8e1dbce3dc5.mp4

We've located the caret position, can determine what text has been typed, and got IPC working on Linux. 🥳 There is still lots to do, but some of the key technical challenges have been solved!

**Comment by LeSnake04** [comment]
Great!

Also good to know Fig is written in rust :)

(The ide logs and tabs showed it)

**Comment by Abhiram4572** [comment]
Waiting!!

**Comment by Pound-Hash** [comment]
> Waiting!!

If you're eager for the app, you can give them a thumbs up or some other similar gesture.

If you're expressing impatience, it's best not to say so.

**Comment by LeSnake04** [comment]
is it like a terminal multiplexer that can be run in any terminal or does it have to be integrated in each terminal.

**Comment by sachinraja** [comment]
@mschrage Could you lock this issue? There are a lot of people subscribed to this who only want to be notified on actual updates.

**Comment by fedeci** [comment]
Post feature requests, ideas and suggestions here: https://github.com/withfig/fig/discussions/1252

**Comment by mschrage** [comment]
Another exciting milestone: we've got the user interface for the autocomplete engine up and running.
https://user-images.githubusercontent.com/3155109/164110187-96d12957-428f-464e-9a36-1fd6285157e5.mp4

The team is making really good progress! We may have some early prototypes ready in the next couple of weeks (assuming everything goes according to plan 🤞).

**Comment by mschrage** [comment]
Fig (basically) working on Fedora using the Gnome desktop environment! 🥳

https://user-images.githubusercontent.com/3155109/167760601-dfbfcc15-3144-4c78-812e-d43b57fc1414.mp4

Fig will be compatible with any terminal on Linux that supports [ibus](https://wiki.archlinux.org/title/IBus). More details to come soon.

**Comment by TheOceanBreeze** [comment]
@mschrage Serious... Which shell are you using here?

**Comment by StarNumber12046** [comment]
Is it usable as of now?

**Comment by grant0417** [comment]
Fig for Linux :penguin: is almost here! We are currently rolling out to Alpha users! :tada: 

If you are interested in testing the Alpha version of Fig for Linux fill out this form: https://withfig.typeform.com/to/OgP8tRrr

**Comment by mtrin** [comment]
@grant0417 Keen to be alpha tester on WSL2 Debian (I presume it's a good test case), sent the form but didn't receive any confirmation but hopefully it worked :)

**Comment by Mahmoudgalalz** [comment]
how can I contribute to this Linux version

**Comment by grant0417** [comment]
Long time coming, but we just launched Linux as part of the successor product Amazon Q! If you want to install [here are the docs](https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-installing.html#command-line-installing-macos).


Additional we just open sourced the codebase so its OSS software as well: https://github.com/aws/amazon-q-developer-cli

**Comment by ameasere** [comment]
> Long time coming, but we just launched Linux as part of the successor product Amazon Q! If you want to install [here are the docs](https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-installing.html#command-line-installing-macos).
> 
> Additional we just open sourced the codebase so its OSS software as well: https://github.com/aws/amazon-q-developer-cli

"Long time coming" - almost 4 years, the project was considered abandoned for a while and the issue threads were locked. Now all of a sudden the project is being revived after being sold by lazy developers to Amazon.

**Comment by yochananmarqos** [comment]
> we just launched Linux 

I don't think that word means what you think it means.


Please archive this repo. It's dead, Jim.

**Comment by ameasere** [comment]
> > we just launched Linux
> 
> I don't think that word means what you think it means.
> 
> Please archive this repo. It's dead, Jim.

Careful, you might get some people giving you a thumbs down for telling the truth!

**Comment by yochananmarqos** [comment]
Breaking news: Truth are lies, lies are truth. Up is down, down is up.

...and now back you your regular distractions...

**Comment by CaidenBlock** [comment]
Fig, as people wanted to actually use it, died years ago.

Archive the repo and cash your check.

**Comment by yochananmarqos** [comment]
/me unsubscribes from this issue I followed almost 4 years ago...

Nevermind the 266 updoots. Nothing to see here folks. Move along....

**Comment by brandondrew** [comment]
wow, why all the hostility?   can anyone explain that to me?

**Comment by CaidenBlock** [comment]
> wow, why all the hostility? can anyone explain that to me?

There was an announcement of a Linux alpha, the features and scope of the project changed dramatically over the next 3 years (not _necessarily_ for the worse), and then you suddenly unlock by saying the corporation you sold your project to has made a Linux port.

This was a piece of software I was interested in, but now it's some fragment of a giant corporation's cloud AI package.

Users who have followed development closely probably see it from a different lens, but those of us who subscribed to an issue 3+ years ago to get updates on a release of software that no longer exists may be less enthused by this "Linux launch".

**Comment by miversen33** [comment]
As an addition, not only were we (users interested in/participating in the Linux Alpha) abandoned due to the sale of this project, but the scope changes of the project made me at the very least uncomfortable with it.

I don't need AI in my CLI. That is not something I asked for. I don't need yet another attempt to inject AI where it doesn't belong. I care for the idea of having a cli tool snooping about in my workspace/system and reporting everything back to "the cloud".

As a final thing, this project is dead. Killed by the developers. I don't blame them for getting paid, but if the project is dead, why are you using an issue here to promote your replacement product _after_ you've already done that?

The only thing this has done is made me remember to unstar and unsub.

I'm happy for you having got paid. Leave me alone

**Comment by theAkito** [comment]
While I think, every project owner is allowed to do whatever the hell he sees fit, including selling it to one of the biggest and most atrocious, disgusting and sickening, (I could go on and on) etc. companies around the globe, pretending, like this project did not die and then continue to promote it here is a different story and deserves a verbal punch in the face.

**Comment by brandondrew** [comment]
> and then you suddenly unlock by saying the corporation you sold your project to has made a Linux port

Me?  Sorry but you've got that part dramatically wrong.

**Comment by brandondrew** [comment]
> Users who have followed development closely probably see it from a different lens, but those of us who subscribed to an issue 3+ years ago to get updates on a release of software that no longer exists may be less enthused by this "Linux launch".

Okay, I read some more about "Amazon Q", but have not tested it (and don't necessarily want to).  It appears that it might (1) try to push you in the direction of using Amazon services in your code, or (2) only give answers related to Amazon services, or (3) require paying for AI through Amazon to work.

But it would be helpful if someone who has followed the project the whole time or tested it or has some deeper knowledge would describe exactly what is wrong rather than merely alluding to it.

Obviously so Jeff Bezos is lowlife slime ball who made literally billions by using his propaganda machine to sow lies and fear about the supposed dangers of Covid and the supposed benefits of the mRNA "vaccine" in order to reap personal rewards, but I doubt that that is the entire reason for the hostility toward Q....

**Comment by alerque** [comment]
This is worse than a license rug pull: the code was never open sourced to begin with so it can't even be forked and continued. The project has an open license but since the actual developed code was never made public and only made available as binaries, it might as well never have existed. Now with the buy-out/sell-out and turning it into a completely different product most of us are **not** interested in, we can't even check out the thing we were interested in in the first place. `"will be open source" != "is open source"`.

**Comment by brandondrew** [comment]
@alerque can you summarize the differences between the original fig and Q?

The claim was made earlier that this is now OSS.  Is that a lie?  Is the Git history not there?  Can't you reconstruct the original fig?

**Comment by alerque** [comment]
@brandondrew No the Git history is not here. It was never posted. The entire repo is a place holder and they were supposedly going to publish the source when they were ready to tag a stable release. Their limited access alpha release process was binary-only and not a source release. In other words the original was never open source and cannot be reconstructed.

The new product seems to be a thin client primarily driven by a proprietary cloud AI service. Who cares if the thin client is open source, the *product* itself is not open sourced or available for tinkering with, extending, changing, etc.

**Comment by brandondrew** [comment]
@alerque Thanks for the added context!

**Comment by CaidenBlock** [comment]
> Me? Sorry but you've got that part dramatically wrong.

Global "you" towards the devs, bad wording.

> Obviously so Jeff Bezos is a lowlife slime ball who made literally billions by using his propaganda machine to sow lies and fear about the supposed dangers of Covid and the supposed benefits of the mRNA "vaccine" in order to reap personal rewards, but I doubt that that is the entire reason for the hostility toward Q....

I'm not angry at Bezos because of some nutjob conspiracy

**Comment by ameasere** [comment]
> @alerque can you summarize the differences between the original fig and Q?
> 
> The claim was made earlier that this is now OSS. Is that a lie? Is the Git history not there? Can't you reconstruct the original fig?

Users having to crawl through Git history, if it was even available, to reconstruct what they once loved should actually tell you a lot about why you're facing so much resistance and hostility with your announcement. It isn't what people wanted, and nobody wants to painstakingly reconstruct what they once loved because the previous owner didn't love it as much.

**Comment by brandondrew** [comment]
@ameasere 

> with your announcement

It is not my announcement.  Do you always make accusations so carelessly?

**Comment by ameasere** [comment]
Notice how you pick the 3 last works of the most meaningless sentence of my reply as well, not the actual point I was making about the feedback you were receiving - which is what you had asked for, might I add. You clearly don't actually care for people to add more context to the "hostility" you claim to receive if you're constantly snapping back at people for adding said context.

**Comment by brandondrew** [comment]
ameasure:

There is nothing in the definition of "accusation" that only applies to crimes.  You called it my announcement, but I didn't make the announcement.  That's very clearly an accusation.  You can claim I'm being hostile but it's clearly you who is hostile.  I was simply trying to understand the hostility here.  While the hostility toward Fig might be warranted, I don't think asking a sincere question warrants any of the hostility that you are directing toward me.

> but the then stupid appendix to his comment became more of a problem

This clearly illustrates my point.

> Notice how you pick the 3 last words of the most meaningless sentence of my reply as well, not the actual point I was making about the feedback you were receiving

Are you not responsible for what you write?  Am I obligated to ignore hostile accusations because you regard them as meaningless?

> You clearly don't actually care for people to add more context to the "hostility" you claim to receive if you're constantly snapping back at people for adding said context.

Pointing out falsehood in your accusation is not "constantly snapping back".  The only person here I see "constantly snapping back" is you.   Do you expect messages of deep gratitude instead?  I was thankful to @alerque for the information he gave.  But your message was quite unlike his.

**Comment by ameasere** [comment]
> ameasure:
> 
> 
> 
> There is nothing in the definition of "accusation" that only applies to crimes.  You called it my announcement, but I didn't make the announcement.  That's very clearly an accusation.  You can claim I'm being hostile but it's clearly you who is hostile.  I was simply trying to understand the hostility here.  While the hostility toward Fig might be warranted, I don't think asking a sincere question warrants any of the hostility that you are directing toward me.
> 
> 
> 
> > but the then stupid appendix to his comment became more of a problem
> 
> 
> 
> This clearly illustrates my point.
> 
> 
> 
> > Notice how you pick the 3 last words of the most meaningless sentence of my reply as well, not the actual point I was making about the feedback you were receiving
> 
> 
> 
> Are you not responsible for what you write?  Am I obligated to ignore hostile accusations because you regard them as meaningless?
> 
> 
> 
> > You clearly don't actually care for people to add more context to the "hostility" you claim to receive if you're constantly snapping back at people for adding said context.
> 
> 
> 
> Pointing out falsehood in your accusation is not "constantly snapping back".  The only person here I see "constantly snapping back" is you.   Do you expect messages of deep gratitude instead?  I was thankful to @alerque for the information he gave.  But your message was quite unlike his.
> 
> 

Yeah I'm not reading all that, happy for you or sorry that happened or something - save the essays for your college teacher, not for dead repositories.

**Comment by brandondrew** [comment]
> Yeah I'm not reading all that, happy for you or sorry that happened or something - save the essays for your college teacher, not for dead repositories.

You demonstrate yet more hostility, and also a disinterest in the facts.

**Comment by ameasere** [comment]
> 
> 
> > Yeah I'm not reading all that, happy for you or sorry that happened or something - save the essays for your college teacher, not for dead repositories.
> 
> 
> 
> You demonstrate yet more hostility, and also a disinterest in the facts.  
> 
> 

You done delivering your TED talk yet?

**Comment by Jeoshua** [comment]
Will you please stop flooding my inbox with this, both of you? God, I don't even remember why I was following this issue...
*Unsubbed*

**Comment by grant0417** [comment]
Hey folks.

First sorry for the brief initial message, I understand the frustration about the long wait for Linux support and the radio silence - that's on us. We should have communicated better during and after the acquisition here.

While the product is now under the Amazon Q brand, the core CLI completion engine remains fundamentally the same as what we originally built Fig around. The AI features that were added are completely optional - you can use the core completion engine without enabling any of them. The 

Regarding open source: We've now released the complete codebase at https://github.com/aws/amazon-q-developer-cli. I know many of you wanted this years ago - we heard you. While it's later than ideal, everything is there now, including the completion engine that many of you were interested. This is fundamentally the same core codebase as Fig, most changes are related to branding and additional (optional) AI features.

I will lock this thread now to prevent email spam for users who would rather not continue to get emailed about this. However if there are any technical details or other concerns you'd like address have feel free to open an issue/discussion in our new repo: https://github.com/aws/amazon-q-developer-cli
